core/rawdb/schema.go

```diff
@@ -188,7 +188,11 @@ func accountSnapshotKey(hash common.Hash) []byte {
 
 // storageSnapshotKey = SnapshotStoragePrefix + account hash + storage hash
 func storageSnapshotKey(accountHash, storageHash common.Hash) []byte {
-	return append(append(SnapshotStoragePrefix, accountHash.Bytes()...), storageHash.Bytes()...)
+	res := make([]byte, 1+len(accountHash)*2)
```

Comment:
```suggestion
	buf := make([]byte, len(trieNodeStoragePrefix)+common.HashLength+len(path))
	n := copy(buf[:], trieNodeStoragePrefix)
	n = copy(buf[n:], accountHash.Bytes())
	copy(buf[n:], path)
	return buf
```
could be something like this.